x/staking/types/msg.go

```diff
@@ -31,12 +31,16 @@ var (
 // NewMsgCreateValidator creates a new MsgCreateValidator instance.
 // Delegator address and validator address are the same.
 func NewMsgCreateValidator(
-	valAddr sdk.ValAddress, pubKey cryptotypes.PubKey, selfDelegation sdk.Coin,
-	description Description, commission CommissionRates, minSelfDelegation sdk.Int,
+	valAddr sdk.ValAddress, pubKey cryptotypes.PubKey, //nolint:interfacer
+	selfDelegation sdk.Coin, description Description, commission CommissionRates, minSelfDelegation sdk.Int,
 ) (*MsgCreateValidator, error) {
-	pkAny, err := codectypes.PackAny(pubKey)
-	if err != nil {
-		return nil, err
+	// TODO: update if we decide to allow nil in NewAnyWithValue
```

Comment:
Did we finally decide to disallow nil? Does that have repercussions in other places in the SDK?

Reply:
Yes, the only impacted places are where we had `PackAny`, because `NewAny*` were panicking on `nil`.

Reply:
BTW, removing the TODO,